miLAG_package/milags_functions.R

```diff
@@ -0,0 +1,1047 @@
+  tryCatch(
+    expr =
+      {nls_LM <- nls_LM(formula = biomass ~ n0 + (time >= lag)*n0*(-1+K*exp(gr_rate*(time-lag))/(K - n0 + n0*exp(gr_rate*(time - lag)))),
+                        data = gr_curve,
+                        start = list(gr_rate = init_gr_rate, K=init_K, lag = init_lag),
+                        control = nls_control(max_iter = max_iter),
+                        lower = lower_bound)
+      },
+    error = function(cond) {
+      nls_LM <<- NA
+    })
+  tryCatch(
+    expr =
+      {nls_LM_no_bound <- nls_LM(formula = biomass ~ n0 + (time >= lag)*n0*(-1+K*exp(gr_rate*(time-lag))/(K - n0 + n0*exp(gr_rate*(time - lag)))),
+                                 data = gr_curve,
+                                 start = list(gr_rate = init_gr_rate, K=init_K, lag = init_lag),
+                                 control = nls_control(max_iter = max_iter))
+      },
+    error = function(cond) {
+      nls_LM_no_bound <<- NA
+    })
+  tryCatch(
+    expr =
+      {nls_PORT <- nls(biomass ~ n0 + (time >= lag)*n0*(-1+K*exp(gr_rate*(time-lag))/(K - n0 + n0*exp(gr_rate*(time - lag)))), gr_curve,
+                        list(gr_rate = init_gr_rate, K=init_K, lag = init_lag),
+                        algorithm = "port",
+                        control = nls_control(max_iter = max_iter),
+                        lower = lower_bound)},
+    error = function(cond) {
+      nls_PORT <<- NA
+    })
+
+  nls <- compare_algorithms(nls_LM_no_bound, nls_PORT, nls_LM)
+  # consider the model without lower bounds only if the resulted estimates are above 0
+  return(nls)
+
+}
+
+
+
+#' calc_lag_fit_to_logistic_with_lag
+#'
+#' Runs nlsLM/nls algorithm of the user's choice to fit the  Logistic model parameters to our data
+#' @param gr_curve data from one specific growth curve with these two columns: time and biomass
+#' @param n0 the initial biomass
+#' @param init_gr_rate initial value for the growth rate
+#' @param init_K initial value for the saturation parameter K
+#' @param init_lag initial value for the lag parameter
+#' @param algorithm defaults to "auto" which chooses between bounded and unbounded Levenberg-Marquardt method and the bounded port method
+#' @param max_iter max. number of iterations; defaults to 100
+#' @param lower_bound lower bound for the bounded nls optimization; defaults to 0
+#' @returns lag and the nls fitting object with parameters fitted to logistic model
+calc_lag_fit_to_logistic_with_lag <- function(gr_curve, n0,
+                                                      init_gr_rate = init_gr_rate,
+                                                      init_K = init_K,
+                                                      init_lag = init_lag,
+                                                      algorithm = "auto",# Levenberg-Marquardt", # or "port" for nls
+                                                      max_iter = 100,
+                                                      lower_bound = c(0,0,0)) {

```

Comment:
sometimes we define functions withh each argument is a separate line, sometimes they are in the same line -> make it homogeneous

Reply:
Done ( only for defining functions, mutate column values are still in multiple lines).